extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -699,11 +834,18 @@ public static void addYouTubeHeaders(final Map<String, List<String>> headers)
     }
 
     /**
-     * Add the <code>X-YouTube-Client-Name</code> and <code>X-YouTube-Client-Version</code> headers.
+     * Add the <code>X-YouTube-Client-Name</code>, <code>X-YouTube-Client-Version</code>,
+     * <code>Origin</code>, and <code>Referer</code> headers.
      * @param headers The headers which should be completed
      */
-    public static void addClientInfoHeaders(final Map<String, List<String>> headers)
+    public static void addClientInfoHeaders(@Nonnull final Map<String, List<String>> headers)
             throws IOException, ExtractionException {
+        if (headers.get("Origin") == null) {
+            headers.put("Origin", Collections.singletonList("https://www.youtube.com"));
+        }
+        if (headers.get("Referer") == null) {
+            headers.put("Referer", Collections.singletonList("https://www.youtube.com"));
+        }
```

Comment:
Done only for the three first headers because I have to add a try catch block which catches `IO` and `Extraction` exceptions for the last header (thank you Java).